network/gossip/libp2p/peerManager_test.go

```diff
@@ -223,14 +223,19 @@ func (suite *PeerManagerTestSuite) TestConcurrentOnDemandPeerUpdate() {
 	// this should trigger the first update and which will block on the ConnectPeers to return
 	unittest.RequireClosesBefore(suite.T(), pm.Ready(), 2*time.Second)
 
+	// assert that the first update started
```

Comment:
there is no guarantee that `peerManager.update` will run after the peer manager has started since GO may just decide not to run that Go routine. Hence, now added a check to first ensure that the `update` was indeed kicked off by checking that `ConnectPeers` was called once).

Reply:
OK, thanks for providing the details. I was thinking do we need to say `return connector.GetNumberOfCalls(t, "ConnectPeers") >= 1` instead of `> 1`

Reply:
at this point there should be `exactly` one call since the call will be blocked by this channel - `connectPeerGate`.

Reply:
cool. Thanks for fixing it 👍